Tests/IntegrationTests/PropagationTests.hpp

```diff
@@ -0,0 +1,477 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/EventData/NeutralTrackParameters.hpp"
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/Propagator/DebugOutputActor.hpp"
+#include "Acts/Surfaces/CylinderSurface.hpp"
+#include "Acts/Surfaces/DiscSurface.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+#include "Acts/Surfaces/StrawSurface.hpp"
+#include "Acts/Tests/CommonHelpers/FloatComparisons.hpp"
+#include "Acts/Utilities/UnitVectors.hpp"
+#include "Acts/Utilities/detail/periodic.hpp"
+
+#include <utility>
+
+// parameter construction helpers
+
+/// Construct (initial) curvilinear parameters.
+inline Acts::CurvilinearParameters makeParametersCurvilinear(double phi,
+                                                             double theta,
+                                                             double absMom,
+                                                             double charge) {
+  using namespace Acts;
+  using namespace Acts::UnitLiterals;
+
+  // phi is ill-defined in forward/backward tracks. normalize the value to
+  // ensure parameter comparisons give correct answers.
+  if (not((0 < theta) and (theta < M_PI))) {
+    phi = 0;
+  }
+
+  Vector3D pos = Vector3D::Zero();
+  double time = 0.0;
+  Vector3D mom = absMom * makeDirectionUnitFromPhiTheta(phi, theta);
+  CurvilinearParameters params(std::nullopt, pos, mom, charge, time);
+
+  // ensure initial parameters are valid
+  CHECK_CLOSE_ABS(params.position(), pos, 0.125_um);
+  CHECK_CLOSE_ABS(params.time(), time, 1_ps);
+  CHECK_CLOSE_ABS(params.momentum(), mom, 0.125_eV);
+  // charge should be identical not just similar
+  BOOST_CHECK_EQUAL(params.charge(), charge);
```

Comment:
I think, I added this because I saw some weird NaNs in #327 and wanted to make sure that the parameters are actually well-defined before I start the propagation.

It does not hurt either and I would just leave it in for now.